Re-add Fun commands!

Wueffi opened the pull request:
This pull request adds the fun commands: /lag /waffle and /nou back to the game.

(Yes Nick this one builds :3)
/nou and /waffle
![image](https://github.com/user-attachments/assets/11d0f3ef-9e35-457b-934f-57c2d46019ff)
Logger + /lag
![image](https://github.com/user-attachments/assets/22ef0af0-53f1-470e-8bfd-d340ed250355)
/lag client side (could capture ingame-message but looks like Original)
![image](https://github.com/user-attachments/assets/58a0cdc3-209c-469d-926f-0ebbb084dfb1)

Accept quik!

Wueffi committed:
Fix Build Process

Wueffi committed:
Add commands

Wueffi committed:
Update config

Wueffi committed:
E!

Wueffi commented:
Planning on Implementing Aliases and other commands latORE

Nickster258 commented:
I'd rather this wait until all other FungeeCommands (The plugin that used to manage `/lag`, `/moo`, and `/c`) commands are added in. Take a look at how [FungeeCommands genericized the commands](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/master/fungee-manager/src/main/kotlin/org/openredstone/fungee/manager/commands/GenericCommand.kt). It [registers the commands](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/master/fungee-manager/src/main/kotlin/org/openredstone/fungee/manager/handlers/DynamicCommandHandler.kt) from the [`commands.json`](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/master/fungee-manager/src/main/resources/commands.json) file.

It would be awesome if we could maintain this `commands.json` format (except convert legacy codes to MiniMessage) and just copy the file over to use in ChattORE. We would also consider having a separate `alias.json` file. This file would have to specify if the alias resides on the proxy or the game server.

Wueffi committed:
Satisfied the needs of the dictator

Wueffi committed:
mm new pr

Wueffi commented:
![image](https://github.com/user-attachments/assets/a1af5ac2-2a58-4be5-9787-524bc0cc0a68)
Yes it builds.

I satisfied the Needs of the nicktator and even added the color list.

e 🧇

Wueffi commented:
![image](https://github.com/user-attachments/assets/8388c480-5c48-4675-86b9-fb9d4976741a)

Wueffi committed:
Very small formatting change

Nickster258 commented on src/main/kotlin/commands/c.kt:
Since we have `commands.json`, I don't think we need a separate `c.kt` for the color command. We can just have that in `commands.json`.

Wueffi replied:
Its because this shouldnt be sent into GlobalChat - adding a localChat parameter is not that necessary if its only one command, but yes we can do that

Nickster258 replied:
Yeah that was how the [OG `/c` was implemented](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/f34c6d97d892c077f82ef19402f9dd65bbeae8ea/fungee-manager/src/main/resources/commands.json#L105).

Nickster258 commented on src/main/kotlin/FunCommands.kt:
```diff
@@ -0,0 +1,103 @@
+        }
+    }
+
+    private fun executeAction(action: String, player: Player) {
```
There *may* be a better way to do this, but I like the idea of having an "allow list" of what command actions can be performed.

However, we may have to reconsider that if we turn this into an aliasing system, but maybe a separate `aliases.json` for that would be best anyway.

Nickster258 commented on build.gradle.kts:
```diff
@@ -1,7 +1,7 @@
 
 plugins {
-    id("com.github.johnrengelman.shadow") version "8.1.1"
+    id("com.gradleup.shadow") version "8.3.0"
```
I didn't realize this was where the shadow plugin went 😱.

Wueffi replied:
xp

Nickster258 reviewed (changes requested):
Only major note is that `/c` could be in `commands.json`. The rest looks fairly straight forward.

Overall I like the PR, but I plan to test it later.

Wueffi committed:
Add localChat support, add c.kt to commands.json 🧇🧐

Wueffi committed:
Merge remote-tracking branch 'refs/remotes/origin/Fun-commands' into Fun-commands

Wueffi commented:
Added localChat Support, moved c.kt into commands.json

Added /op-me and /funcommands
![image](https://github.com/user-attachments/assets/3e2922f6-5943-4738-93a0-00b8300d6ddb)

Nickster258 commented on src/main/kotlin/FunCommands.kt:
```diff
@@ -0,0 +1,109 @@
+
+            if (commandConfig.globalChat != null) {
+                chattORE.broadcast(renderedMessage)
+            } else if (commandConfig.localChat != null) {
```
I think it'd be nice if it could do both local and global chat if they were set, not one or the other. [The old FungeeCommands allowed for both to occur at the same time](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/f34c6d97d892c077f82ef19402f9dd65bbeae8ea/fungee-manager/src/main/kotlin/org/openredstone/fungee/manager/commands/GenericCommand.kt#L29).

Nickster258 reviewed (changes requested):
Just one note that the old "spec" allowed for both global and local chat events to happen in a command. I think it would be good to keep that for this.

Wueffi committed:
Added dual Chat Support

Wueffi commented:
Yea uh i call it DUAL CHAT SUPPORT

Wueffi committed:
mm forgot a } dotn judge

Wueffi committed:
Reviewing own PR xd [Last small fix this is perfect now]
